src/app/services/shared-state-app.service.ts

```diff
@@ -2,13 +2,12 @@ import { Injectable } from '@angular/core';
 import { SignalsManager } from '../utilities/signals-manager.utility';
 import { User } from '../models';
 
-export enum AppSignalKeys {
-  'USER' = 'user',
-  'TEST' = 'test',
-}
+export const AppSignalKeys = {
```

Comment:
Perfect !!! <3 as const is really good for this cases, it will tell Typescript to infer the most specific and narrow type as possible which in return gives us this great case of use for typing dynamic solutions